### PR TITLE
Don't ask for email when reporting issues

### PR DESCRIPTION
**Pull request by MatMoore**
https://github.com/ministryofjustice/find-moj-data/issues/995

Instead of requiring the reporter to enter their email, we can just use the email of the currently signed in user.

Instead, just ask if they would like to be emailed.

I'm using a TypedChoiceField instead of a BooleanField, because BooleanField has weird `required` behaviour. The docs encourage you to set `required=False` if you need to be able to submit a False value, but this means you can't validate that a value was set if you use a RadioWidget.

### CODE REVIEW

File: feedback/models.py [hunk 1]
```diff
@@ -34,6 +36,10 @@ class IssueChoices(models.TextChoices):
     created = models.DateTimeField(auto_now_add=True)
     modified = models.DateTimeField(auto_now=True)
 
+    created_by = models.ForeignKey(
+        CustomUser, on_delete=models.CASCADE, blank=False, null=True
```
Comment (mitchdawson1982) [inline feedback/models.py:40]: Might be worth using the `get_user_model()` helper to refer to the User model. https://docs.djangoproject.com/en/5.1/topics/auth/customizing/#django.contrib.auth.get_user_model
``
from django.contrib.auth import get_user_model

created_by = models.ForeignKey(
        get_user_model(), on_delete=models.CASCADE, blank=False, null=True
)
``

File: tests/feedback/conftest.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import pytest
+
+from users.models import CustomUser
```
Comment (mitchdawson1982) [inline tests/feedback/conftest.py:3]: Might be worth using the `get_user_model()` helper to save referring to the User model directly. https://docs.djangoproject.com/en/5.1/topics/auth/customizing/#django.contrib.auth.get_user_model
```
from django.contrib.auth import get_user_model

user = get_user_model()
user.objects.create(email="reporter@justice.gov.uk")
```

Comment (MatMoore) [reply tests/feedback/conftest.py:3]: that's a lot nicer, thanks. I'll refactor this once I've got the thing working

File: feedback/forms.py [hunk 3]
```diff
@@ -53,3 +50,13 @@ class Meta:
...
         formfield_callback = formfield
+
+    send_email_to_reporter = forms.TypedChoiceField(
+        widget=RadioSelect(
+            attrs={"class": "govuk-radios__input", "required": "True"},
+        ),
+        choices=(("Yes", _("Yes")), ("No", _("No"))),
+        coerce=lambda x: x == "Yes",
+        required=True,
+        label=_("Do you want us to email you a copy of this report?"),
```
Comment (MatMoore) [inline feedback/forms.py:61]: not sure about this copy - let me know if you have something better

Comment (mitchdawson1982) [reply feedback/forms.py:61]: Perhaps - _Would you like us to email you a copy of this report?_

**Review by mitchdawson1982 — DISMISSED** [review]
LGTM

**Review by mitchdawson1982 — APPROVED** [review]
LGTM

**Pull request merged**